packages/contracts/test/routerFactory.spec.ts

```diff
@@ -0,0 +1,109 @@
+import { ethers, waffle } from "hardhat";
+import { expect, use } from "chai";
+import { solidity } from "ethereum-waffle";
+
+use(solidity);
+
+import { Wallet, providers, Contract, constants } from "ethers";
+import { TransactionManager } from "@connext/nxtp-contracts";
+import TransactionManagerArtifact from "@connext/nxtp-contracts/artifacts/contracts/TransactionManager.sol/TransactionManager.json";
+import RouterArtifact from "@connext/nxtp-contracts/artifacts/contracts/Router.sol/Router.json";
+
+import { deployContract, assertReceiptEvent } from "./utils";
+
+// import types
+import { RouterFactory } from "../typechain";
+
+import { getContractError } from "../src";
+
+const { AddressZero } = constants;
+const createFixtureLoader = waffle.createFixtureLoader;
+describe("Router Factory Contract", function () {
```

Comment:
coverage is broken, and needs to be improved for the factory and router contracts